src/engine/pieces/pawn.ts

```diff
@@ -1,13 +1,49 @@
 import Piece from './piece';
 import Player from '../player';
 import Board from '../board';
+import Square from '../square';
 
 export default class Pawn extends Piece {
     public constructor(player: Player) {
         super(player);
     }
 
     public getAvailableMoves(board: Board) {
-        return new Array(0);
+        const currentPosition = board.findPiece(this);
+        let moveList = [];
+        
+        if (this.player === Player.WHITE){
+            const candidateMove1 = Square.at(currentPosition.row+1, currentPosition.col);
+            if (!board.isPieceAtSquare(candidateMove1)){
+
+                moveList.push(Square.at(currentPosition.row+1, currentPosition.col));
+                
+                if (currentPosition.row === 1) {
+                    const candidateMove2 = Square.at(currentPosition.row+2, currentPosition.col);
+
+                    if (!board.isPieceAtSquare(candidateMove2)){
+                        moveList.push(Square.at(currentPosition.row+2, currentPosition.col));
```

Comment:
```suggestion
                        moveList.push(candidateMove2);
```

and same applies for the black player